tests/agents/handagent/conftest.py

```diff
@@ -50,4 +50,6 @@ def env(request):
 
     shared_env.reset()
 
-    return shared_env
+    yield shared_env
+
+    shared_env.__on_exit__()
```

Comment:
Great! Can we include some kind of `kill` or `exit` function in `HolodeckEnvironment`? I don't think we should be calling `__on_exit__` directly. It just sounds like an event handler that shouldn't be called from external modules. Looks like it's the current norm in the project. Thoughts?

Reply:
Looking into this, it looks like we can [just use `with` syntax](https://pytest.readthedocs.io/en/2.9.1/yieldfixture.html), which is much cleaner. It calls `HolodeckEnvironment.__exit__()`, which calls `__on_exit__`.  I'll make those changes in a few minutes.